profiler/src/ProfilerEngine/Datadog.Linux.ApiWrapper/functions_to_wrap.c

```diff
@@ -306,25 +317,40 @@ void initLibrary(void)
                 //  - dotnet run sets DOTNET_DbgEnableMiniDump=1
                 //  - dotnet then launches the target app
                 //  - the target app thinks DOTNET_DbgEnableMiniDump has been set by the user and enables passthrough
-                setenv(DD_INTERNAL_CRASHTRACKING_PASSTHROUGH, "1", 1);
+                real_setenv(DD_INTERNAL_CRASHTRACKING_PASSTHROUGH, "1", 1);
             }
         }
         else
         {
             // If DOTNET_DbgEnableMiniDump is not set, we set it so that the crash handler is called,
             // but we instruct it to not call createdump afterwards
-            setenv(COMPlus_DbgEnableMiniDump, "1", 1);
-            setenv(DOTNET_DbgEnableMiniDump, "1", 1);
-            setenv(DD_INTERNAL_CRASHTRACKING_PASSTHROUGH, "0", 1);
+            real_setenv(COMPlus_DbgEnableMiniDump, "1", 1);
+            real_setenv(DOTNET_DbgEnableMiniDump, "1", 1);
+            real_setenv(DD_INTERNAL_CRASHTRACKING_PASSTHROUGH, "0", 1);
+        }
+
+        originalMiniDumpName = real_getenv(DOTNET_DbgMiniDumpName);
+
+        if (originalMiniDumpName == NULL || strncmp(originalMiniDumpName, datadogCrashMarker, strlen(datadogCrashMarker)) == 0)
+        {
+            originalMiniDumpName = real_getenv(COMPlus_DbgMiniDumpName);
+        }
+
+        if (originalMiniDumpName != NULL && strncmp(originalMiniDumpName, datadogCrashMarker, strlen(datadogCrashMarker)) == 0)
+        {
+            // If LD_PRELOAD was set in the parent process, then we replaced COMPlus_DbgMiniDumpName with datadogCrashMarker and lost the original value
+            // We use DD_INTERNAL_CRASHTRACKING_MINIDUMPNAME to retrieve it
+            originalMiniDumpName = real_getenv(DD_INTERNAL_CRASHTRACKING_MINIDUMPNAME);
         }
 
-        originalMiniDumpName = getenv(DOTNET_DbgMiniDumpName);
-        if (originalMiniDumpName == NULL)
+        if (originalMiniDumpName != NULL && originalMiniDumpName[0] != '\0')
         {
-            originalMiniDumpName = getenv(COMPlus_DbgMiniDumpName);
+            // Save the original value in DD_INTERNAL_CRASHTRACKING_MINIDUMPNAME so that child processes can retrieve it
+            real_setenv(DD_INTERNAL_CRASHTRACKING_MINIDUMPNAME, originalMiniDumpName, 1);
         }
-        setenv(COMPlus_DbgMiniDumpName, datadogCrashMarker, 1);
-        setenv(DOTNET_DbgMiniDumpName, datadogCrashMarker, 1);
+
+        real_setenv(COMPlus_DbgMiniDumpName, datadogCrashMarker, 1);
+        real_setenv(DOTNET_DbgMiniDumpName, datadogCrashMarker, 1);
```

Comment:
Probably a silly question, but is there a reason we have to clobber and restore the original values? Why can't we just store our marker in the DD_INTERNAL_ env instead?